src/Utils/Path.idr

```diff
@@ -153,196 +159,234 @@ lexPath str = let (tokens, _, _, _) = lex pathTokenMap str in tokens
+||| Some kind of invalid paths are accepted. The relax rules:
 |||
-||| - Both slash('/') and backslash('\\') are parsed as valid directory
-|||   separator, regardless of the platform;
-||| - Any characters in path body in allowed, e.g., glob like "/root/*";
-||| - Verbatim prefix(`\\?\`) that disables the forward
-|||   slash (Windows only) is ignored.
-||| - Repeated separators are ignored, so "a/b" and "a//b" both have "a" 
+||| - Both slash('/') and backslash('\\') are parsed as valid directory separator,
+|||   regardless of the platform;
+||| - Any characters in the body in allowed, e.g., glob like "/root/*";
+||| - Verbatim prefix(`\\?\`) that disables the forward slash is ignored (for
+|||   Windows only).
+||| - Repeated separators are ignored, therefore, "a/b" and "a//b" both have "a"
 |||   and "b" as bodies.
-||| - Occurrences of "." are normalized away, except if they are at the
-|||   beginning of the path. For example, "a/./b", "a/b/", "a/b/". and 
-|||   "a/b" all have "a" and "b" as bodies, but "./a/b" starts with an 
-|||   additional `CurDir` body.
+||| - "." in the body are removed, unless they are at the beginning of the path.
+|||   For example, "a/./b", "a/b/", "a/b/." and "a/b" will have "a" and "b" as
+|||   bodies, and "./a/b" will starts with `CurDir`.
 |||
 ||| ```idris example
 ||| parse "C:\\Windows/System32"
-||| ```
-||| ```idris example
 ||| parse "/usr/local/etc/*"
 ||| ```
 export
 parse : String -> Path
-parse str = case parse parsePath (lexPath str) of
-                 Right (p, _) => p
-                 _ => emptyPath
+parse str =
+  case parse parsePath (lexPath str) of
+    Right (path, _) => path
+    _ => emptyPath
 
 --------------------------------------------------------------------------------
 -- Utils
 --------------------------------------------------------------------------------
 
 isAbsolute' : Path -> Bool
-isAbsolute' p = if isWindows
-                  then case p.volume of
-                            Just (UNC _ _) => True
-                            Just (Disk _) => p.hasRoot
-                            Nothing => False
-                  else p.hasRoot
+isAbsolute' path =
+  if isWindows then
+    case path.volume of
+      Just (UNC _ _) => True
+      Just (Disk _) => path.hasRoot
+      Nothing => False
+  else
+    path.hasRoot
 
 append' : (left : Path) -> (right : Path) -> Path
-append' l r = if isAbsolute' r || isJust r.volume
-                 then r
-                 else if hasRoot r
-                         then record { volume = l.volume } r
-                         else record { body = l.body ++ r.body,
-                                       hasTrailSep = r.hasTrailSep } l
+append' left right =
+  if isAbsolute' right || isJust right.volume then
+    right
+  else if hasRoot right then
+    record { volume = left.volume } right
+  else 
+    record { body = left.body ++ right.body, hasTrailSep = right.hasTrailSep } left
+
+splitPath' : Path -> List Path

```

Comment:
Since the only use of this function is in `splitPath`, why not return `List String` instead of going through `map show` to get the components? Semantically it feels very odd to return a list of paths which are all singleton paths

Reply:
It's originally for avoiding to repeat the login in `Show`, i.e. showing the root and give trailSep to the last body.

Reply:
Alright! I've taken a look and I couldn't come up with something significantly better. If something's wrong with it we can always fix it in a subsequent pull request